feat(alerting): Add Incident.io alerting provider (#972)
TwiN/gatus: README.md

| `alerting.gitlab`          | Configuration for alerts of type `gitlab`. <br />See [Configuring GitLab alerts](#configuring-gitlab-alerts).                           | `{}`    |
| `alerting.googlechat`      | Configuration for alerts of type `googlechat`. <br />See [Configuring Google Chat alerts](#configuring-google-chat-alerts).             | `{}`    |
| `alerting.gotify`          | Configuration for alerts of type `gotify`. <br />See [Configuring Gotify alerts](#configuring-gotify-alerts).                           | `{}`    |

> | `alerting.incidentio`      | Configuration for alerts of type `incidentio`. <br />See [Configuring Incident.io alerts](#configuring-incidentio-alerts).              | `{}`    |

Reviewer: this should be incident-io as well.

Author: done.

> | `alerting.incident-io`     | Configuration for alerts of type `incidentio`. <br />See [Configuring Incident.io alerts](#configuring-incidentio-alerts).              | `{}`    |

Reviewer: ```suggestion
| `alerting.incident-io`     | Configuration for alerts of type `incident-io`. <br />See [Configuring Incident.io alerts](#configuring-incidentio-alerts).              | `{}`    |
```

| `alerting.telegram`        | Configuration for alerts of type `telegram`. <br />See [Configuring Telegram alerts](#configuring-telegram-alerts).                     | `{}`    |
| `alerting.twilio`          | Settings for alerts of type `twilio`. <br />See [Configuring Twilio alerts](#configuring-twilio-alerts).                                | `{}`    |
| `alerting.zulip`           | Configuration for alerts of type `zulip`. <br />See [Configuring Zulip alerts](#configuring-zulip-alerts).                              | `{}`    |

> | `alerting.incidentio`           | Configuration for alerts of type `incidentio`. <br />See [Configuring Incident.io alerts](#configuring-incidentio-alerts).                              | `{}`    |

Reviewer: Can you move this line so that it's sorted alphabetically?

Author: done.

![Gotify notifications](.github/assets/gotify-alerts.png)

#### Configuring Incident.io alerts
| Parameter                          | Description                                                                                | Default       |
|:-----------------------------------|:-------------------------------------------------------------------------------------------|:--------------|

> | `alerting.incidentio`                   | Configuration for alerts of type `incidentio`                                                   | `{}`          |

Reviewer: Should be incident-io here as well

Author: Done.

> | `alerting.incidentio.alert-source-config-id`       | Which alert source config produced this alert                                                                         | Required `""` |

Reviewer 2: **thoughts**: I've had a look at an implementation few days ago before I saw your message and had this thought.

`alert-source-config-id` is mentioned in the API spec, but nowhere else.

When "Connecting your HTTP source to incident.io", we have the following UI:
![image](https://github.com/user-attachments/assets/c926fc1f-7760-414a-bedc-81b587dc2910)

I'd think usability-wise, it would look good to offer `url` and `bearer_token` or something of the sort as a configuration option.

Author: done.

| `alerting.incidentio.auth-token`     | Token that is used for authentication. |  Required `""`          |

> | `alerting.incidentio.overrides`         | List of overrides that may be prioritized over the default configuration                   | `[]`          |

Reviewer 2: Then, we'd expect the alerts created by such a config to have the following fields defined, as per the API:
```yaml
{
   // ...
  "status": "firing",
  "source_url": "https://my-gatus.my-domain.com",
  "metadata": {
    "team": "very-core",
    "service": "very-api"
  }
}
```
With when we define this config:
```yaml
alerting:
  incidentio:
    # ...
    metadata:
      team: very-core
      service: very-api
    source-url: https://my-gatus.my-domain.com
    default-alert:
      send-on-resolved: true
```

> | `alerting.incidentio.overrides`         | List of overrides that may be prioritized over the default configuration                   | `[]`          |

Author: Yeah, probably should add them as well since they integrate into the workflow of alerting on incident.io

> | `alerting.incidentio.overrides`         | List of overrides that may be prioritized over the default configuration                   | `[]`          |

Author: Done.

| `alerting.incidentio.default-alert`     | Default alert configuration. <br />See [Setting a default alert](#setting-a-default-alert) | N/A           |
| `alerting.incidentio.overrides[].group` | Endpoint group for which the configuration will be overridden by this configuration        | `""`          |
| `alerting.incidentio.overrides[].*`     | See `alerting.incidentio.*` parameters                                                          | `{}`          |

```yaml

> alerting:
>   incident-io:
>      alert-source-config-id: "*****************"
>         auth-token: "********************************************"

Reviewer 3: ```suggestion
alerting:
  incident-io:
    alert-source-config-id: "*****************"
    auth-token: "********************************************"
```

Author: as @TwiN requested, it should be incident-io, but forgot to update the docs to match the yaml example, thanks.

Author: if you have the time, and willing, changing the docs to incident-io insted of incidentio(I'm currently not near a computer)

Author: good catch!

endpoints:
  - name: website
    url: "https://twin.sh/health"
    interval: 30s
    conditions:
      - "[STATUS] == 200"
      - "[BODY].status == UP"
      - "[RESPONSE_TIME] < 300"
    alerts:
      - type: incident-io
        description: "healthcheck failed"
        send-on-resolved: true
```
in order to get the required alert source config id and authentication token, you must configure an HTTP alert source.

> > **_NOTE:_**  the source config id is of the form { api.incident.io/v2/alert_events/http/{CONFIG-SOURCE-ID}}
> 
> > **_NOTE:_** the auth token is of the form { "Authorization": "Bearer {AUTH-TOKEN}" }

Reviewer: ```suggestion
> **_NOTE:_**  the source config id is of the form `api.incident.io/v2/alert_events/http/$ID` and the token is expected to be passed as a bearer token like so: `Authorization: Bearer $TOKEN`

> **_NOTE:_** ```

| `alerting.incidentio`                   | Configuration for alerts of type `incidentio`                                                   | `{}`          |
| `alerting.incidentio.alert-source-config-id`       | Which alert source config produced this alert                                                                         | Required `""` |
| `alerting.incidentio.auth-token`     | Token that is used for authentication. |  Required `""`          |
| `alerting.incidentio.overrides`         | List of overrides that may be prioritized over the default configuration                   | `[]`          |
| `alerting.incidentio.default-alert`     | Default alert configuration. <br />See [Setting a default alert](#setting-a-default-alert) | N/A           |
| `alerting.incidentio.overrides[].group` | Endpoint group for which the configuration will be overridden by this configuration        | `""`          |
| `alerting.incidentio.overrides[].*`     | See `alerting.incidentio.*` parameters                                                          | `{}`          |

```yaml
alerting:

>   incidentio:

Reviewer: Can you rename this to incident-io?

Author: done.

![Slack notifications](.github/assets/slack-alerts.png)

> 

Reviewer: ```suggestion
```
